spring-shell2-shell1-adapter/src/main/java/org/springframework/shell2/legacy/LegacyParameterResolver.java

```diff
@@ -82,7 +91,58 @@ public Object resolve(MethodParameter methodParameter, List<String> words) {
 
 	@Override
 	public ParameterDescription describe(MethodParameter parameter) {
-		throw new UnsupportedOperationException();
+		Parameter jlrParameter = parameter.getMethod().getParameters()[parameter.getParameterIndex()];
+		CliOption option = jlrParameter.getAnnotation(CliOption.class);
+		ParameterDescription result = ParameterDescription.outOf(parameter);
+		result.help(option.help());
+		List<String> keys = Arrays.asList(option.key());
+		result.keys(keys.stream()
+				.filter(key -> !key.isEmpty())
+				.map(key -> CLI_PREFIX + key)
+				.collect(Collectors.toList()));
+		Optional<String> defaultValue = defaultValueFor(option, result.keys());
+		if (defaultValue.isPresent()) {
+			result.defaultValue(defaultValue.get());
+		}
+		boolean containsEmptyKey = keys.contains("");
+		result.mandatoryKey(!containsEmptyKey);
```

Comment:
Hmmm, taking that back.

In the `sum2` example, this is what we get:

```
	@CliCommand(value = "sum2", help = "adds two numbers")
	public int sum2(
			@CliOption(key = "v1", unspecifiedDefaultValue = "38") int a,
			@CliOption(key = "v2", specifiedDefaultValue = "42", mandatory = true) int b,
			@CliOption(key = "v3", mandatory = false) int c) {
		return a + b + c;
	}
	
help sum2


NAME
	sum2 - adds two numbers

SYNOPSYS
	sum2 [--v1 int]  [--v2 int]  [--v3 int]

OPTIONS
	--v1  int
		  [Optional, default = 38]

	--v2  int
		  [Optional, default = 42 if used as --v2]

	--v3  int
		  [Optional, default = null]

```

Having `v2` marked as mandatory, I _think_ it would make sense to have the synopsys say
`sum2 [--v1 int]  --v2 [int]  [--v3 int]`. Likewise, in the `OPTIONS` section, the word "Optional" may be misleading